src/plugins/intel_cpu/tests/functional/shared_tests_instances/snippets/add.cpp

```diff
@@ -14,8 +14,15 @@ namespace {
 
 namespace snippets_static_1 {
 // These  inputs are needed to test static Loop optimizations (emit the whole tile, body with increments, set WA etc)
-std::vector<ov::Shape> inShapesStatic1{{1, 16, 29,  1}, {1, 16, 29,  7}, {1, 16, 29,  8}, {1, 16, 29,  15}, {1, 16, 29,  16}, {1, 16, 29,  31}};
-std::vector<ov::Shape> inShapesStatic2{{1, 16, 29,  1}, {1, 16, 1, 1}, {1, 1, 1, 1}};
+std::vector<ov::test::InputShape> inShapesStatic1{{{}, {{1, 16, 29, 1}}},
+                                                  {{}, {{1, 16, 29, 7}}},
+                                                  {{}, {{1, 16, 29, 8}}},
+                                                  {{}, {{1, 16, 29, 15}}},
+                                                  {{}, {{1, 16, 29, 16}}},
+                                                  {{}, {{1, 16, 29, 31}}}};
+std::vector<ov::test::InputShape> inShapesStatic2{{{}, {{1, 16, 29, 1}}},
+                                                  {{}, {{1, 16, 1, 1}}},
+                                                  {{}, {{1, 1, 1, 1}}}};
```

Comment:
UPD: just discussed the issue with @dmitry-gorokhov. He suggests to go with the implemented approach and to rewrite all static tests to use InputShape's. The motivation behind that is to avoid tests class duplication for dynamic cases.

Reply:
snippets test is updated with InputShape. Dynamic test case is added.
Note: Not cover MHA, MatMul related as recompilation to matmul takes long time, and critical for LLM customers. Team is seeking shape agnostic MHA solution.

Reply:
Maybe there is sense to use `static_shapes_to_test_representation` for static shape tests to avoid a lot of `{{}}`. What do you think?

Reply:
The unified paramSet structure with
[inputShape(using InputShape = std::pair<ov::PartialShape, std::vector<ov::Shape>>;)](https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/shared_test_classes/include/shared_test_classes/base/ov_subgraph.hpp#L16)
is shared for static and dynamic test cases. If define two paramSet structure, seems like we need two setup() and getName(). I am not sure it's deserve to do it. Maybe we can use static_shapes_to_test_representation for test only include static test cases. Thank you Alexandra!